custom_components/tuya_local/devices/petoneer_fresco_ezgo_petfountain.yaml

```diff
@@ -0,0 +1,124 @@
+name: Pet fountain
+products:
+  - id: jfyxbljcu1lz99hg
+    manufacturer: Petoneer
+    model: Fresco EzGo
+entities:
+  - entity: switch
+    icon: "mdi:water-pump"
+    dps:
+      - id: 1
+        type: boolean
+        name: switch
+  - entity: select
+    name: Mode
+    category: config
+    dps:
+      - id: 2
+        type: string
+        name: option
+        mapping:
+          - dps_val: normal
+            value: normal
+          - dps_val: smart
+            value: geyser
+  - entity: sensor
+    name: Filter cleaning in
+    class: duration
+    category: diagnostic
+    dps:
+      - id: 3
+        type: integer
+        name: sensor
+        unit: d
+  - entity: sensor
+    name: Pump cleaning in
+    class: duration
+    category: diagnostic
+    dps:
+      - id: 4
+        type: integer
+        name: sensor
+        unit: d
+  - entity: button
+    name: Reset filter cleaning timer
+    category: diagnostic
+    dps:
+      - id: 5
+        type: boolean
+        name: button
+        optional: true
+  - entity: button
+    name: Reset pump cleaning timer
+    category: diagnostic
+    dps:
+      - id: 6
+        type: boolean
+        name: button
+        optional: true
+  - entity: sensor
+    name: Water level
+    class: enum
+    category: diagnostic
+    dps:
+      - id: 9
+        name: sensor
+        type: string
+        mapping:
+          - dps_val: "0"
+            value: "No water"
+          - dps_val: "1"
+            value: "OK"
+          - dps_val: "2"
+            value: "Low"
+          - dps_val: "3"
+            value: "Pump not detected"
+          - dps_val: "4"
+            value: "Pump clogged"
+  - entity: light
+    class: indicator
+    category: config
+    dps:
+      - id: 12
+        type: integer
+        name: switch
+        mapping:
+          - dps_val: 0
+            value: false
+          - dps_val: 1
+            value: true
+  - entity: binary_sensor
+    name: Error
```

Comment:
Remove the name to let the class translations show through.

Generally also, I would include a "fault_code" attribute, with the raw value of dp 23.
I'd also suggest separating routine things out from the maintenance issues to their own binary_sensor (dps_val: 1 especially, there is a tank_empty translation key defined for such binary_sensors).

Reply:
It's kind of weird, this mostly overlaps `water_level`, which is why I kept it like that.